### PR TITLE
Add function for #154

### PR DESCRIPTION
**Pull request by Robinlovelace**

### CODE REVIEW

**Comment by agila5** [comment]
Hi Robin! Thanks for the PR. I checked the review and I think it's quite positive 🎉. Anyway, I'm a little bit busy in the next days but I will work on osmextract during the weekend (or, anyway, as soon as possible).

**Comment by Robinlovelace** [comment]
This is ready for review @agila5, no rush but it seems to work. Output is very chatty atm which is no bad thing I think, feedback welcome on idea and implementation:

```
ghent = oe_get("ghent")
Warning: The input place was matched with multiple geographical zones: Ghana - Kent. Selecting the first match.
No exact matching found for place = ghent. Best match is Ghana.
String distance between best match and the input place is 2, while the maximum threshold distance is equal to 1. You can try increasing the max_string_dist parameter, look for a closer match in the chosen provider database or consider using a different match_by variable.
No match in the OSM provider data. Searching for the location online
trying URL 'https://nominatim.openstreetmap.org/search?q=ghent&limit=1&format=geojson'
downloaded 591 bytes

Got data for: Gent, Oost-Vlaanderen, België / Belgique / Belgien, c(3.7250121, 51.0538286)
although coordinates are longitude/latitude, st_intersects assumes that they are planar
The input place was matched with multiple geographical areas. Selecting the areas with the highest "level". See the help page associated to the chosen provider for an explanation of the meaning of the "level" field
trying URL 'https://download.geofabrik.de/europe/belgium-latest.osm.pbf'
Content type 'application/octet-stream' length 418022088 bytes (398.7 MB)
==================================================
downloaded 398.7 MB

File downloaded!
Start with the vectortranslate operations on the input file!
0...10...20...30...40...50...60...70...80...90...100 - done.
Finished the vectortranslate operations on the input file!
Reading layer `lines' from data source `/mnt/57982e2a-2874-4246-a6fe-115c199bc6bd/data/osm/geofabrik_belgium-latest.gpkg' using driver `GPKG'
Simple feature collection with 1486676 features and 9 fields
geometry type:  LINESTRING
dimension:      XY
bbox:           xmin: -5.652017 ymin: 49.43529 xmax: 8.589302 ymax: 62.51466
geographic CRS: WGS 84
```

**Comment by agila5** [comment]
Hi @Robinlovelace! Checking it right now. I was thinking of a slightly different workflow, but I will add some code for testing/discussing in a few hours.

**Review by agila5 — APPROVED** [review]
Hi @Robinlovelace! I just pushed a commit for this PR. I slightly changed your approach in two ways: 1) the `oe_search()` function now is in `oe_match()` since it could be useful outside of `oe_get()` and 2) `oe_match()` now checks the other providers before using `oe_serch()`. So, for example, 

``` r
library(osmextract)
oe_match("Leeds")
#> No exact match found for place = Leeds and provider = geofabrik. Best match is Laos. 
#> Checking the other providers.
#> I found an exact string match using provider = bbbike so I'm going to return that.
#> $url
#> [1] "https://download.bbbike.org/osm/bbbike/Leeds/Leeds.osm.pbf"
#> 
#> $file_size
#> [1] 19376705
oe_match("Olginate")
#> No exact match found for place = Olginate and provider = geofabrik. Best match is Suriname. 
#> Checking the other providers.
#> No exact match found in any OSM provider data. Searching for the location online.
#> although coordinates are longitude/latitude, st_intersects assumes that they are planar
#> The input place was matched with multiple geographical areas. Selecting the areas with the highest "level". See the help page associated to the chosen provider for an explanation of the meaning of the "level" field
#> $url
#> [1] "https://download.geofabrik.de/europe/italy/nord-ovest-latest.osm.pbf"
#> 
#> $file_size
#> [1] 416306623
```

<sup>Created on 2020-12-28 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

Now I need to review examples and vignettes and add some tests

**Comment by Robinlovelace** [comment]
> Hi @Robinlovelace! I just pushed a commit for this PR. I slightly changed your approach in two ways: 1) the `oe_search()` function now is in `oe_match()` since it could be useful outside of `oe_get()` and 2) `oe_match()` now checks the other providers before using `oe_serch()`. 

Great thinking on including it in `oe_match()` :+1:

**Comment by Robinlovelace** [comment]
Another thought: worth using `oe_match()` in `oe_download()` so that if you give it something other than a URL it still tries to find something useful?

**Comment by agila5** [comment]
> Another thought: worth using oe_match() in oe_download() so that if you give it something other than a URL, it still tries to find something useful?

I would like to keep the two operations in two distinct functions since if you want just to download PBF/GPKG data for a given provider, then you can run `oe_get(input, download_only = TRUE)` (adding `skip_vectortranslate` if you are not interested in the GPKG conversion). If you want, we can create a wrapper around `oe_get` with those parameters as default, not sure about the name.

**Comment by agila5** [comment]
Sometimes next week I will finish re-reading the vignette and then I think it will be ready for review/merge!

**Comment by Robinlovelace** [comment]
I think this is almost ready to go. Let me know if you want me to add something to the vignette. I've already used the new functionality in my work, useful for sure!

**Comment by agila5** [comment]
Hi Robin! Thanks for the comment. If you can test this PR and find any problem, please report it. I plan to finish the PR on Monday 11/01 and then start with #160

**Comment by Robinlovelace** [comment]
Thanks @agila5, sounds good to me!

**Comment by agila5** [comment]
Hi @Robinlovelace, I think it's ready for review.

**Comment by agila5** [comment]
Hi @Robinlovelace, sorry to bother you. Can you check the PR today or tomorrow morning? Tomorrow (after lunch) I would like to complete the other PRs and write the answer for rOpenSci review.

File: R/download.R [hunk 1]
```diff
@@ -36,10 +36,10 @@
 #' @export
 #'
 #' @examples
-#' its_match = oe_match("ITS Leeds", provider = "test")
+#' its_match = oe_match("ITS Leeds", provider = "test", quiet = TRUE)
```
Comment (Robinlovelace) [inline R/download.R:39]: :+1:

File: R/get.R [hunk 2]
```diff
@@ -116,18 +112,24 @@
...
 #'
 #' \dontrun{
 #' # A more complex example
 #' west_yorkshire = oe_get("West Yorkshire", quiet = FALSE)
-#' # If you run it again, the function will not download the file or convert it
+#' # If you run it again, the function will not download the file
+#' # or convert it again
 #' west_yorkshire = oe_get("West Yorkshire", quiet = FALSE)
+#' # Match with place name
+#' oe_get("Milan") # Warning: the .pbf file is 400MB
```
Comment (Robinlovelace) [inline R/get.R:129]: Great example!

File: R/match.R [hunk 3]
```diff
@@ -265,36 +274,87 @@ oe_match.character = function(
...
       )
     }
-    if (interactive() && isTRUE(interactive_ask)) {
-      continue = utils::menu(
-        choices = c("Yes", "No"),
-        title = "Do you confirm that this is the right match?"
+
+    # 2. Check the other providers and, if there is an exact match, just return
+    # the matched value from the other provider:
+    other_providers = setdiff(oe_available_providers(), provider)
```
Comment (Robinlovelace) [inline R/match.R:292]: :+1:

File: R/match.R [hunk 4]
```diff
@@ -367,7 +427,7 @@ oe_match_pattern = function(
 
   # Then we extract only the elements of the match_by_column that match the
   # input pattern.
-  match_ID = grep(pattern, match_by_column)
+  match_ID = grep(pattern, match_by_column, ignore.case = TRUE)
```
Comment (Robinlovelace) [inline R/match.R:430]: :+1:

File: R/zzz.R [hunk 5]
```diff
@@ -1,10 +1,12 @@
 .onAttach = function(libname, pkgname) {
   packageStartupMessage(paste(
     "Data (c) OpenStreetMap contributors, ODbL 1.0. https://www.openstreetmap.org/copyright.",
-    "Any product made from OpenStreetMap must cite OSM as the data source.",
-    "Geofabrik data are taken from https://download.geofabrik.de/",
-    "For usage details of bbbike data see https://download.bbbike.org/osm/",
-    "OpenStreetMap_fr data are taken from http://download.openstreetmap.fr/",
+    "Check the package website for more details.",
```
Comment (Robinlovelace) [inline R/zzz.R]: One suggestion: Add the link to the website.

**Review by Robinlovelace — COMMENTED** [review]
Looks good!

**Pull request merged**